network/p2p/gossip/gossipable.go

```diff
@@ -8,8 +8,12 @@ import "github.com/ava-labs/avalanchego/ids"
 // Gossipable is an item that can be gossiped across the network
 type Gossipable interface {
 	GetID() ids.ID
-	Marshal() ([]byte, error)
-	Unmarshal(bytes []byte) error
+}
+
+// Marshaller creates concrete Gossipable types from bytes
+type Marshaller[T Gossipable] interface {
+	GossipMarshal(T) ([]byte, error)
+	GossipUnmarshal([]byte) (T, error)
```

Comment:
I feel like this reads better:
```suggestion
	MarshalGossip(T) ([]byte, error)
	UnmarshalGossip([]byte) (T, error)
```